test.js

```diff
@@ -162,118 +157,168 @@ describe('DNS Endpoint Pool', function () {
-      var resolve = autoRestore(Sinon.stub(DEP.prototype, 'resolve')),
-          errorHandler = Sinon.spy(),
-          errObj = { error: true },
-          dep;
+    describe('using percentage error rates', function () {
+       var ejectOnErrorConfig = {
+        failureRate: 1,
+        failureRateWindow: 2,
+        resetTimeout: 10000
+      };
 
-      // works on the first and fourth calls, fails every other time
-      resolve
-        .callsArgWith(0, errObj)
-        .onFirstCall().callsArgWith(0, null, [
-          { name: 'bar.localhost', port: 8000 },
-          { name: 'baz.localhost', port: 8001 }
-        ])
-        .onCall(3).callsArgWith(0, null, [
+      it('will remove endpoints from the pool if they fail', function () {
+        var resolve = autoRestore(Sinon.stub(DEP.prototype, 'resolve'));
+
+        resolve.callsArgWith(0, null, [
           { name: 'bar.localhost', port: 8000 },
           { name: 'baz.localhost', port: 8001 }
         ]);
 
-      dep = new DEP('foo.localhost', 5000, ejectOnErrorConfig);  // call 1
-      dep.on('updateError', errorHandler);
-      clock.tick(5000); // call 2
-      clock.tick(5000); // call 3
-      clock.tick(5000); // call 4, should reset the timer
-      clock.tick(5000); // call 5
+        var dep = new DEP('foo.localhost', 5000, ejectOnErrorConfig);
+
+        var barEndpoint = dep.getEndpoint();
+        barEndpoint.callback(true);
 
-      Sinon.assert.calledThrice(errorHandler);
+        var bazEndpoint = dep.getEndpoint();
 
-      expect(errorHandler.firstCall.calledWithExactly(errObj, 5000)).to.be(true);
-      expect(errorHandler.secondCall.calledWithExactly(errObj, 10000)).to.be(true);
-      expect(errorHandler.thirdCall.calledWithExactly(errObj, 5000)).to.be(true);
-      dep.stopUpdating();
+        expect(dep.getEndpoint()).to.be(barEndpoint); // still in the pool
+        barEndpoint.callback(true);
+
+        expect(dep.getEndpoint()).to.be(bazEndpoint);
+        expect(dep.getEndpoint()).to.be(bazEndpoint); // bar is removed
+
+        var status = dep.getStatus();
+        expect(status.total).to.be(2);
+        expect(status.unhealthy).to.be(1);
+        expect(status.age).to.be.a('number');
+
+        dep.stopUpdating();
+      });
+      it('will remove an endpoint from the pool once it reaches its error threshold', function () {
+        var resolve = autoRestore(Sinon.stub(DEP.prototype, 'resolve'));
+
+        resolve.callsArgWith(0, null, [
+          { name: 'bar.localhost', port: 8000 }
+        ]);
+        var dep = new DEP('foo.localhost', 5000, {
+          failureRate: 0.75,
+          failureRateWindow: 4,
+          resetTimeout: 10000
+        });
+        var endpoint = dep.getEndpoint();
+        endpoint.callback(true);
+        endpoint = dep.getEndpoint();
+        endpoint.callback(true);
+        endpoint = dep.getEndpoint();
+        endpoint.callback(true);
+        expect(dep.getEndpoint(), 'to be', null);
```

Comment:
It does if you use it - when we invent a way to auto-require it in all projects everywhere, we will have achieved world domination :)